src/GeneratorUtils.php

```diff
@@ -1,22 +1,32 @@
 <?php
 
 namespace CommandString\Utils;
-use InvalidArgumentException;
 
 class GeneratorUtils {
-    public static function uuid(int $length = 16, ?array $characters = null): string
+    public static function uuid(int $length = 16): string
     {
-        $characters = $characters ?? array_merge(range("A", "Z"), range("a", "z"), range(0, 9));
-        $id = "";
+        $salt = random_bytes($length);
+        $time = microtime(true);
+        $hash = hash('sha256', $salt . $time);
+        return $hash;
+    }
 
-        if (count($characters) < 2) {
-            throw new InvalidArgumentException("The character array must have two items!");
+    public static function randomString(int $length = 16, ?array $characters = null): string
+    {
+        if (is_null($characters)) {
+            $characters = array_merge(
+                range('a', 'z'),
+                range('A', 'Z'),
+            );
         }
 
+        $charactersLength = count($characters) - 1;
+        $string = '';
+
         for ($i = 0; $i < $length; $i++) {
-            $id .= $characters[rand(0, count($characters)-1)];
+            $string .= $characters[random_int(0, $charactersLength)];
         }
 
-        return $id;
+        return $string;
```

Comment:
What not foreach the $characters array again? Should be using [mt_rand](https://www.php.net/manual/en/function.mt-rand.php). Also, why was the exception removed?

~~I understand this probably was my original code.~~

Reply:
> What not foreach

can you rephrase? I'm confused?

Reply:
Oh wait I'm sorry I had a moment. Ignore the foreach suggestion